exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
-    verify(node).submitTransaction(eq(expectedRawTx));
+  void submitCreateCounter() {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      String counterName = "bids";
+      service.submitCreateCounter(counterName);
+      testKit.createBlock();
+
+      testKit.withSnapshot((view) -> {
+        QaSchema schema = new QaSchema(view);
+        MapIndex<HashCode, Long> counters = schema.counters();
+        MapIndex<HashCode, String> counterNames = schema.counterNames();
+        HashCode counterId = sha256().hashString(counterName, UTF_8);
+
+        assertThat(counters.get(counterId)).isEqualTo(0L);
+        assertThat(counterNames.get(counterId)).isEqualTo(counterName);
+        return null;
+      });
+    }
   }
 
   @Test
-  void submitIncrementCounter() throws Exception {
-    setServiceNode(node);
-    when(node.submitTransaction(any(RawTransaction.class))).thenReturn(DEFAULT_TX_MESSAGE_HASH);
-
-    long seed = 1L;
-    HashCode counterId = sha256()
-        .hashString("Cats counter", UTF_8);
-    HashCode txHash = service.submitIncrementCounter(seed, counterId);
-
-    IncrementCounterTx expectedTx = new IncrementCounterTx(seed, counterId);
-    RawTransaction expectedRawTx = IncrementCounterTx.converter().toRawTransaction(expectedTx);
+  void submitIncrementCounter() {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      String counterName = "bids";
+      service.submitCreateCounter(counterName);
+      testKit.createBlock();
 
-    assertThat(txHash).isEqualTo(DEFAULT_TX_MESSAGE_HASH);
-    verify(node).submitTransaction(eq(expectedRawTx));
+      HashCode counterId = sha256().hashString(counterName, UTF_8);
+      service.submitIncrementCounter(1L, counterId);
+      testKit.createBlock();
+      testKit.withSnapshot((view) -> {
+        QaSchema schema = new QaSchema(view);
+        MapIndex<HashCode, Long> counters = schema.counters();
+        MapIndex<HashCode, String> counterNames = schema.counterNames();
+
+        assertThat(counters.get(counterId)).isEqualTo(1L);
+        assertThat(counterNames.get(counterId)).isEqualTo(counterName);
+        return null;
+      });
+    }
   }
 
   @Test
-  void submitValidThrowingTx() throws Exception {
-    setServiceNode(node);
-    when(node.submitTransaction(any(RawTransaction.class))).thenReturn(DEFAULT_TX_MESSAGE_HASH);
+  void submitValidThrowingTx() {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      service.submitValidThrowingTx(1L);
+      testKit.createBlock();
 
-    long seed = 1L;
-    HashCode txHash = service.submitValidThrowingTx(seed);
+      List<String> logMessages = logAppender.getMessages();
```

Comment:
No, it is not acceptable to test like that. This code must verify that a _QA service_
submits a _certain_ transaction. This test, however, verifies that an implementation of TransactionAdapter, given the log configuration of this module, adds a certain log message.